pkg/helmraiser/helm.go

```diff
@@ -0,0 +1,152 @@
+package helmraiser
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	jsonnet "github.com/google/go-jsonnet"
+	"github.com/google/go-jsonnet/ast"
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v3"
+)
+
+func confToArgs(conf map[string]interface{}) ([]string, []string, error) {
+	var args []string
+	var tempFiles []string
+
+	// create file and append to args
+	if val, ok := conf["values"]; ok {
+		if len(val.(map[string]interface{})) > 0 {
+			valuesYaml, err := yaml.Marshal(val.(interface{}))
+			if err != nil {
+				return nil, nil, err
+			}
+			tmpFile, err := ioutil.TempFile(os.TempDir(), "tanka-")
+			if err != nil {
+				return nil, nil, errors.Wrap(err, "cannot create temporary values.yaml")
+			}
+			tempFiles = append(tempFiles, tmpFile.Name())
+			if _, err = tmpFile.Write(valuesYaml); err != nil {
+				return nil, tempFiles, errors.Wrap(err, "failed to write to temporary values.yaml")
+			}
+			if err := tmpFile.Close(); err != nil {
+				return nil, tempFiles, err
+			}
+			args = append(args, fmt.Sprintf("--values=%s", tmpFile.Name()))
+		}
+	}
+
+	// append custom flags to args
+	if val, ok := conf["flags"]; ok {
```

Comment:
From a Jsonnet perspective, one would most likely not want to pass flags, but config parameters in some form.

It feels more natural to the language to do:

```jsonnet
chart: helm.template("grafana", "grafana", {
  opts: {
    "include-crds": true
  },
})
```
This is logically no different from the current implementation, only the conversion of "Jsonnet types" to flags needs to happen elsewhere (Go-code or Jsonnet intermediate code)

Reply:
Given some of these flags could even break helmraiser if passed (output-dir for example), it could even make sense to select those we want to support and define a struct instead of a general `map[string]interface{}`

Reply:
I like the idea, can you add it to the design doc?

Reply:
I don't want to implement it now, having the ability to play with the raw flags at this stage of development is what I want, I'm not sure if we can cover every flag to a map[string]interface{} (for example `--is-upgrade` is only a string{}, not a map.

Reply:
Yeah the fixed struct is probably overkill at this stage, agreed.

A map based model however maintains all flexibility. The only difference is that instead of passing flags, these are specified in an object. Your mentioned `is-upgrade` is actually a bool flag, so it would be set to true. 

Consumer code would need to detect this and change it to a proper flag. Such code is already present in kausal `mapToFlags`, otherwise trivial to implement in Go

Reply:
Agreed in another PR to not do this right now, but reconsider before 1.0